### PR TITLE
feat(share-video) Allow sharing direct video links (mp4 etc) on mobile

### PR DESCRIPTION
**Pull request by quitrk**

### CODE REVIEW

**Comment by quitrk** [comment]
Jenkins please test this please

File: android/sdk/src/main/java/org/jitsi/meet/sdk/ReactInstanceManagerHolder.java [hunk 1]
```diff
@@ -190,6 +191,7 @@ static void initReactInstanceManager(Activity activity) {
                 new com.reactnativecommunity.webview.RNCWebViewPackage(),
                 new com.rnimmersive.RNImmersivePackage(),
                 new com.zmxv.RNSound.RNSoundPackage(),
+                new ReactVideoPackage(),
```
Comment (saghul) [inline android/sdk/src/main/java/org/jitsi/meet/sdk/ReactInstanceManagerHolder.java:194]: Please use the full package name here, as we do for other third-party packages.

File: ios/Podfile.lock [hunk 2]
```diff
@@ -559,7 +567,12 @@ SPEC CHECKSUMS:
   react-native-netinfo: 8d8db463bcc5db66a8ac5c48a7d86beb3b92f61a
   react-native-slider: b733e17fdd31186707146debf1f04b5d94aa1a93
   react-native-splash-screen: 200d11d188e2e78cea3ad319964f6142b6384865
+<<<<<<< HEAD
   react-native-webrtc: 2e8095a43dd3d95ce796d9750280c0f48aadad3d
+=======
```
Comment (saghul) [inline ios/Podfile.lock:572]: Oops?

File: ios/Podfile.lock [hunk 3]
```diff
@@ -580,6 +593,10 @@ SPEC CHECKSUMS:
   RNWatch: a5320c959c75e72845c07985f3e935e58998f1d3
   Yoga: 96b469c5e81ff51b917b92e8c3390642d4ded30c
 
+<<<<<<< HEAD
 PODFILE CHECKSUM: 1fa5a1e259f145d32c1ca968b26dac65cff34b49
+=======
```
Comment (saghul) [inline ios/Podfile.lock:598]: Ditto.

File: ios/app/src/main.m [hunk 4]
```diff
@@ -20,7 +20,7 @@
 
 int main(int argc, char * argv[]) {
     @autoreleasepool {
-        return UIApplicationMain(
+          return UIApplicationMain(
```
Comment (saghul) [inline ios/app/src/main.m:23]: Unnecessary change.

File: react/features/shared-video/components/native/AbstractVideoManager.js [hunk 5]
```diff
@@ -0,0 +1,288 @@
...
+            return;
+        }
+
+        const playerTime = await this.getTime();
+
+        if (shouldSeekToPosition(_time, playerTime)) {
+            this.seek(_time);
+        }
+
+        if (this.getPlaybackState() !== _status) {
```
Comment (saghul) [inline react/features/shared-video/components/native/AbstractVideoManager.js:153]: Can we be a bit more consistent here and call both either "status" or "state" ?

File: react/features/shared-video/components/native/AbstractVideoManager.js [hunk 6]
```diff
@@ -0,0 +1,288 @@
...
+            if (_status === PLAYBACK_STATES.PLAYING) {
+                this.play();
+            }
+
+            if (_status === PLAYBACK_STATES.PAUSED) {
```
Comment (saghul) [inline react/features/shared-video/components/native/AbstractVideoManager.js:158]: Else if?

File: react/features/shared-video/components/native/AbstractVideoManager.js [hunk 7]
```diff
@@ -0,0 +1,288 @@
...
+
+/**
+ * Maps part of the props of this component to Redux actions.
+ *
+ * @param {Function} dispatch - The Redux dispatch function.
+ * @returns {Props}
+ */
+export function _mapDispatchToProps(dispatch: Function): $Shape<Props> {
+    return {
+        _setSharedVideoStatus: ({ videoUrl, status, time, ownerId }) => {
```
Comment (saghul) [inline react/features/shared-video/components/native/AbstractVideoManager.js:279]: Generally, we've found `mapDispatchToProps` not to be very useful. It's a lot of boilerplate for a one liner function call here. Do you mind just dispatch()-ing the action directly instead?

File: react/features/shared-video/components/native/SharedVideo.js [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+     */
+    render() {
+        const {
+            isOwner,
+            playerHeight,
+            playerWidth,
+            videoUrl
+        } = this.props;
+
+        if (!videoUrl) {
```
Comment (saghul) [inline react/features/shared-video/components/native/SharedVideo.js:111]: When can this happen? If it is null, should we still call `setWideScreenMode` in any case?

File: react/features/shared-video/components/native/SharedVideo.js [hunk 9]
```diff
@@ -0,0 +1,170 @@
...
+            return null;
+        }
+
+        return (
+            <View
+                pointerEvents = { isOwner ? 'auto' : 'none' }
+                style = { styles.youtubeVideoContainer } >
```
Comment (saghul) [inline react/features/shared-video/components/native/SharedVideo.js:118]: Can we rename this style while we're at it?

File: react/features/shared-video/components/native/SharedVideo.js [hunk 10]
```diff
@@ -0,0 +1,170 @@
...
+                {videoUrl.match(/http/)
```
Comment (saghul) [inline react/features/shared-video/components/native/SharedVideo.js:119]: I think we might want to make this a bit more future-proof my turning it into an object where we only populate a key based on the video type:

```
{
    videoUrl: "https://lalala.mp4"
}
```

```
{
    youtubeId: "mjndnkoldj901"
}
````

This way if we add more sources like Vimeo or what have you we should be able to support them easier, WDYT?

File: react/features/shared-video/components/native/SharedVideoDialog.js [hunk 11]
```diff
@@ -22,40 +21,12 @@ class SharedVideoDialog extends AbstractSharedVideoDialog<*> {
         return (
             <InputDialog
                 contentKey = 'dialog.shareVideoTitle'
-                onSubmit = { this._onSetVideoLink }
+                onSubmit = { super._onSetVideoLink }
```
Comment (saghul) [inline react/features/shared-video/components/native/SharedVideoDialog.js:24]: While testing I'm getting an error unless I bind this function to `this`.

**Review by saghul — CHANGES_REQUESTED** [review]
Left some comments, PTAL.

Tangentially related, I noticed the YT sharing doesn't seem to work properly, I tried to share Big Buck Bunny and it says the video is unavailable :-/ Tried with some other, same thing.

Not related to this PR though, but maybe you can take a look since you are here.

File: react/features/shared-video/components/native/AbstractVideoManager.js (hunk 6 above)
Comment (quitrk) [reply react/features/shared-video/components/native/AbstractVideoManager.js:158]: i considered they're already mutually exclusive so it won't matter

**Comment by quitrk** [comment]
looks like an update to the youtube lib fixed sharing issue

File: react/features/shared-video/components/native/SharedVideo.js (hunk 10 above)
Comment (quitrk) [reply react/features/shared-video/components/native/SharedVideo.js:119]: i think we need to think about this outside the scope of this PR since it most likely involves breaking backwards compatibility with older clients

File: react/features/shared-video/components/native/SharedVideo.js (hunk 8 above)
Comment (quitrk) [reply react/features/shared-video/components/native/SharedVideo.js:109]: when stopping shared video, the shared video state is reset before the fake video participant is removed

**Review by saghul — APPROVED** [review]
LGTM, excellent job! Thanks a lot for fixing YT sharing too! ❤️

**Comment by saghul** [comment]
Looks like the test failure is relevant: http://pr-reports.jitsi.org/jitsi-meet-PR-test/9886/chrome-chrome-reports/html/

**Comment by quitrk** [comment]
Jenkins please test this please
